packages/@aws-cdk/aws-cloudfront/lib/distribution.ts

```diff
@@ -1,14 +1,332 @@
+        throw new Error('\'responseCode\' must be provided if \'responsePagePath\' is defined');
+      }
+      if (!errorResponse.responseCode && !errorResponse.errorCachingMinTtl) {
+        throw new Error('A custom error response without either a \'responseCode\' or \'errorCachingMinTtl\' is not valid.');
+      }
+    }
+    this.errorConfigurations.forEach(e => validateCustomErrorResponse(e));
+
+    return this.errorConfigurations.map(errorConfig => {
+      return {
+        errorCachingMinTtl: errorConfig.errorCachingMinTtl?.toSeconds(),
+        errorCode: errorConfig.errorCode,
+        responseCode: errorConfig.responseCode,
+        responsePagePath: errorConfig.responsePagePath,
+      };
+    });
+  }
+
+}
+
+/**
+ * The price class determines how many edge locations CloudFront will use for your distribution.
+ */
+export enum PriceClass {
+  PRICE_CLASS_100 = 'PriceClass_100',
+  PRICE_CLASS_200 = 'PriceClass_200',
+  PRICE_CLASS_ALL = 'PriceClass_All'
+}
+
+/**
+ * How HTTPs should be handled with your distribution.
+ */
+export enum ViewerProtocolPolicy {
+  HTTPS_ONLY = 'https-only',
+  REDIRECT_TO_HTTPS = 'redirect-to-https',
+  ALLOW_ALL = 'allow-all'
+}
+
+/**
+ * Defines what protocols CloudFront will use to connect to an origin.
+ */
+export enum OriginProtocolPolicy {
+  HTTP_ONLY = 'http-only',
+  MATCH_VIEWER = 'match-viewer',
+  HTTPS_ONLY = 'https-only',
+}
+
+/**
+ * The HTTP methods that the Behavior will accept requests on.
+ */
+export class AllowedMethods {
+  /** HEAD and GET */
+  public static readonly ALLOW_GET_HEAD = new AllowedMethods(['GET', 'HEAD']);
+  /** HEAD, GET, and OPTIONS */
+  public static readonly ALLOW_GET_HEAD_OPTIONS = new AllowedMethods(['GET', 'HEAD', 'OPTIONS']);
+  /** All supported HTTP methods */
+  public static readonly ALLOW_ALL = new AllowedMethods(['GET', 'HEAD', 'OPTIONS', 'PUT', 'PATCH', 'POST', 'DELETE']);
+
+  /** HTTP methods supported */
+  public readonly methods: string[];
+
+  private constructor(methods: string[]) { this.methods = methods; }
+}
+
+/**
+ * Options for configuring custom error responses.
+ *
+ * @experimental
+ */
+export interface CustomErrorResponse {
+  /**
+   * The minimum amount of time, in seconds, that you want CloudFront to cache the HTTP status code specified in ErrorCode.
+   *
+   * @default the default caching TTL behavior applies
+   */
+  readonly errorCachingMinTtl?: Duration;
+  /**
+   * The HTTP status code for which you want to specify a custom error page and/or a caching duration.
+   */
+  readonly errorCode: number;
+  /**
+   * The HTTP status code that you want CloudFront to return to the viewer along with the custom error page.
+   *
+   * If you specify a value for `responseCode`, you must also specify a value for `responsePagePath`.
+   *
+   * @default not set, the error code will be returned as the response code.
+   */
+  readonly responseCode?: number;
```

Comment:
```suggestion
  readonly responseHttpStatus?: number;
```

Reply:
Sure; I admit I'm still getting a feel for when we want to use existing service names for concepts, and when we are comfortable renaming.

Reply:
I we can relax this guidelines when this is about generic terms (like "http status"). The important semantics to preserve are for *service features*: I wouldn't want to rename "lambda layers" to "lambda dependencies" just because it makes more sense because then users won't be able to bind these terms to the service documentation.